gateway-ha/src/test/java/io/trino/gateway/CustomImageNameSubstitutor.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway;
+
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.ImageNameSubstitutor;
+
+public class CustomImageNameSubstitutor
+        extends ImageNameSubstitutor
+{
+    private static final String TESTCONTAINERS_TRINO_IMAGE_SUBSTITUTE = "TESTCONTAINERS_TRINO_IMAGE_SUBSTITUTE";
+
+    @Override
+    public DockerImageName apply(DockerImageName dockerImageName)
+    {
+        if (dockerImageName.asCanonicalNameString().equals("trinodb/trino:latest") && System.getenv(TESTCONTAINERS_TRINO_IMAGE_SUBSTITUTE) != null) {
+            String imageSubstitute = System.getenv(TESTCONTAINERS_TRINO_IMAGE_SUBSTITUTE);
+            return DockerImageName.parse(imageSubstitute).asCompatibleSubstituteFor("trinodb/trino");
+        }
+        return dockerImageName;
+    }
+
+    @Override
+    protected String getDescription()
+    {
+        return "custom image name substitutor";
+    }
+}
```

Comment:
`testcontainers.properties` isn't required?

Reply:
i was going to use env variables to configure this. https://java.testcontainers.org/features/configuration/
It looks like we can just choose one

Reply:
If we use env variables, why this class is required? I'm looking the following sentence:

> Setting environment variables TESTCONTAINERS_HUB_IMAGE_NAME_PREFIX=registry.mycompany.com/mirror/
https://java.testcontainers.org/features/image_name_substitution/#automatically-modifying-docker-hub-image-names

Reply:
this may not work because it needs image name to be predictable

Reply:
I was looking at this https://java.testcontainers.org/features/image_name_substitution/#developing-a-custom-function-for-transforming-image-names-on-the-fly

Reply:
> it needs image name to be predictable

I don't understand this. Can you elaborate on that?

Reply:
> If we use env variables, why this class is required? I'm looking the following sentence:
> 
> > Setting environment variables TESTCONTAINERS_HUB_IMAGE_NAME_PREFIX=registry.mycompany.com/mirror/
> > https://java.testcontainers.org/features/image_name_substitution/#automatically-modifying-docker-hub-image-names

The link you are looking at is specific to add prefix to image name. It has an assumption:

>Consider this if:
>>Your private registry has copies of images from Docker Hub where the names are predictable, and just adding a prefix is enough. For example, registry.mycompany.com/mirror/mysql:8.0.36 can be derived from the original Docker Hub image name (mysql:8.0.36) with a consistent prefix string: registry.mycompany.com/mirror/

Also, it looks like even though we can change the registry, the names of the image have to match. In our case, we have different image names from the dockerhub.

Reply:
Thanks for your explanation. 

> we have different image names from the dockerhub.

This is the important fact I didn't know. Please leave the code comment.

Reply:
added.

Reply:
> added.

I can't find it.